### PR TITLE
Add docker for production

### LINKED ISSUE #1815: Feature Request: Create Docker Image for "production" use
### Summary
This feature request proposes the creation and provision of an Docker image for phpvms. The introduction of an Docker image will streamline the setup process for users, offering a more convenient and standardized environment for running the software, alongside the existing development setup facilitated by Laravel Sail.

### Context
Currently, phpvms supports two primary methods of setup:

1. Development environment using Laravel Sail.
2. Bare-metal setup as described in the project documentation.

While these methods are functional, the absence of an Docker image presents a barrier for users who prefer or require containerized environments for deployment and management. An Docker image would bridge this gap, enhancing the usability and adoption of phpvms.

### Benefits

- **Ease of Setup**: Users can quickly set up a phpvms environment without dealing with complex dependencies and configurations.
- **Consistency**: The Docker image ensures a consistent environment across different machines and setups, reducing the "it works on my machine" problem.
- **Portability**: Containerized environments can be easily moved across different systems, facilitating testing, development, and deployment.

### Conclusion
Providing an Docker image for phpvms will significantly enhance the user experience by simplifying the setup process and providing a reliable, consistent, and portable environment for managing virtual airlines. This addition aligns with modern development and deployment practices, ensuring phpvms remains accessible and easy to use for a broader audience.

### PR DESCRIPTION
**Pull request by arthurpar06**
Closes #1815 

**STILL WIP**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM dunglas/frankenphp
```
Comment (Lapotor) [inline Dockerfile:1]: A image version should be defined. Otherwise mismatching PHP versions could occur.

File: docker-compose.prod.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      - internal
+    depends_on:
+      - mariadb
+      - redis
+
+  mariadb:
+    image: 'mariadb:11'
+    restart: unless-stopped
+    ports:
+      - '${FORWARD_DB_PORT:-3306}:3306'
```
Comment (Lapotor) [inline docker-compose.prod.yml:22]: We don't need to expose ports if the DB will only be used for the app.

File: docker-compose.prod.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        - ping
+        - '-p${DB_PASSWORD}'
+      retries: 3
+      timeout: 5s
+
+  redis:
+    image: 'redis:alpine'
+    restart: unless-stopped
+    ports:
+      - '${FORWARD_REDIS_PORT:-6379}:6379'
```
Comment (Lapotor) [inline docker-compose.prod.yml:46]: We don't need to expose ports unless we want to use redis outside from this stack.

**Comment by arthurpar06** [comment]
Hi @Lapotor,
Thanks for your feedback. I'll be pushing quite a few things in the coming days. I've worked a lot on it to get cron, queue workers, etc. I'll switch my production over to it, and after a few days of testing, I'll update this PR.

**Comment by Slyplayer8990** [comment]
Should be approved, it is important for getting a development environment as easy as possible.

**Comment by nabeelio** [comment]
> Should be approved, it is important for getting a development environment as easy as possible.

Looks like a few things still need to be addressed

**Comment by arthurpar06** [comment]
Alright, so I've been running a Docker stack on my server for a while, but it has some drawbacks, and I'm not skilled enough with Docker to fix everything. I need your opinions.

For me, there should be 4 containers:
- Our container built from the Dockerfile to handle web requests
- Our container built from the Dockerfile to handle tasks (cron + queues, including Laravel Horizon for queues)
- MariaDB
- Redis

A few things I think need to be revisited:

1. If a user has modules mounted in `/modules,` they need to run `composer dump-autoload` for the classes to be loaded at launch. However, I don’t know how to modify the app container's entrypoint while still keeping the default entrypoint afterward (I haven’t managed to figure out the command that the default entrypoint runs).

2. Is it really a good practice to have two containers from the same image with two different entrypoints for tasks (queues/cron) and for the web application? I don’t know Docker well enough to understand what the best practices are in this case.

3. If the user has multiple apps on their server, they would then have a reverse proxy. Is it a good idea to have a reverse proxy that then points to the Caddy/FrankenPHP instance of this application? That’s what I use in production, but is there a better way to do it?

**Comment by arthurpar06** [comment]
Ultimately, we’re not going to use FrankenPHP in order to have something more stable and robust with PHP-FPM, see #1914

**Pull request closed without merging**